numpy/lib/tests/test_function_base.py

```diff
@@ -2716,6 +2820,28 @@ def test_with_incorrect_minlength(self):
                             "must not be negative",
                             lambda: np.bincount(x, minlength=-1))
 
+        out = np.zeros(5, dtype=int)
+        assert_raises_regex(ValueError,
+                            "cannot be larger than the size of 'out'",
+                            lambda: np.bincount(x, minlength=10, out=out))
+
+        initial = np.zeros(5, dtype=int)
+        assert_raises_regex(ValueError,
+                            "cannot be larger than the size of 'initial'",
+                            lambda: np.bincount(x, minlength=10, 
+                                                initial=initial))
+
+    def test_with_incorrect_types(self):
+        x = np.array([], dtype=int)
+        out = np.empty(6, dtype=float)
+        assert_raises_regex(TypeError,
+                            "Cannot cast",
+                            lambda: np.bincount(x, out=out))
+        initial = np.empty(6, dtype=float)
+        assert_raises_regex(TypeError,
+                            "Cannot cast",
+                            lambda: np.bincount(x, initial=initial))
```

Comment:
Here the case is a bit less clear, since one clearly could cast (if unsafely). This would be especially nice since one could then really just define a default of zero for the argument (`initial=0`), which would cast to `float` when `weight` is present.